qhana_plugin_registry/api/models/generators/env.py

```diff
@@ -68,12 +68,14 @@ def generate_link(
         if query_params is None:
             query_params = {}
 
+        scheme = current_app.config.get("PREFERRED_URL_SCHEME", "http")
```

Comment:
Without these changes requests over https returns http URLs:
![image](https://user-images.githubusercontent.com/2245391/235673502-4606c130-68eb-4bba-bac1-d09e549e55e9.png)
![image](https://user-images.githubusercontent.com/2245391/235673600-486af49c-22cb-4548-ad69-0f977ddb0b7f.png)
Also, `PREFERRED_URL_SCHEME` was not used by flask automatically in this case, that's why I manually set the `_scheme` argument everywhere.
The reverse proxy already is configured to redirect `http` to `https`, but Chrome doesn't even try to connect to `http` urls when your on a `https` website. Firefox automatically used `https` instead of `http`.

Reply:
Oh, I know what the problem is here. It can be somewhat tricky to deploy a wsgi app https-aware. By default proxy servers do not forward headers that indicate that the original request was over https. The reverse proxy must set the relevant headers:

e.g. for an Apache config:
```ini
# this header forwards the used scheme
    RequestHeader set X-Forwarded-Proto "https,https"
# this header forwards the used hostname of the proxy server (should be set automatically by the reverse proxy)
    RequestHeader set X-Forwarded-For "qhana-registry.example.com"
```

Reply:
I'm using Caddy as reverse proxy and according to the documentation it automatically sets the these headers. But it seems like flask needs to be configured to use them: https://flask.palletsprojects.com/en/2.3.x/deploying/proxy_fix/. I will try that.

Reply:
Yes, trusting arbitrary reverse proxy headers can be a security risk.